eldritch/events.py

```diff
@@ -1256,6 +1256,8 @@ def finish_str(self):
 
 def Draw(character, deck, draw_count, prompt="Choose a card", keep_count=1, target_type=None):
   cards = DrawItems(character, deck, draw_count, target_type=target_type)
+  if keep_count == 'all':
```

Comment:
Do we want to support this? Better question: in what circumstances do you imagine this would be used? I could imagine this being deceptive when combined with someone like the archaeologist or salesman. Or maybe it's not deceptive because it would do exactly the right thing?

Reply:
In the expansions, there are encounters like: 
> Pass a [Speed (-1) check](http://www.arkhamhorrorwiki.com/Speed_check) to draw [Unique items](http://www.arkhamhorrorwiki.com/Unique_item) equal to your successes, keeping any [Tomes](http://www.arkhamhorrorwiki.com/Tome) that you draw and discarding the rest. ([Celeano](http://www.arkhamhorrorwiki.com/Celeano_(encounters)))

or 

> Make a [Luck (-2) check](http://www.arkhamhorrorwiki.com/Luck_check). If you succeed, draw 3 [Unique Item](http://www.arkhamhorrorwiki.com/Unique_Item) cards and keep any that cost $4 or less. Discard the rest. ([Curiositie Shoppe](http://www.arkhamhorrorwiki.com/Curiositie_Shoppe))

Reply:
I believe how this would work with the Archaeologist is that you would pass in `keep_count="all"`, but the archaeology ability would replace the original `DrawItems`  with a `DrawItems` that has one higher for DrawCount.  However, since `keep_count` is being set in the `Draw` function, they would only get to keep the original number.  We would, additionally, need to add something like a `keep_criteria` kwarg to KeepDrawn to check that the cards pass (e.g. `keep_criteria=lambda c: isinstance(c, Tome)` or `keep_criteria=lambda c: c.cost <= 4`)

Reply:
Alternate to keep_criteria: create a FilterDrawn event. Use a Sequence([DrawItems, FilterDrawn, KeepDrawn]), where FilterDrawn exposes the same interface as DrawItems. I don't know yet if I like that idea better; it depends on how messy KeepDrawn is going to become.